### PR TITLE
Budget App Capstone Project PR

### PR DESCRIPTION
**Pull request by YoutLone**
### I implement the project according to the project requirements. It is 

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in.
- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Make unit and integration tests for all the most important components of my RoR application.

 ### _Thank you for your review🙏🙏_

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
```
Comment (mukhammadrizooff) [inline app/views/devise/registrations/edit.html.erb:1]: - [ ] [REQUIRED] Good job so far! Kindly your `Log In` page has difference between [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), there are `Navbar` should doesn't appear and `Sign Up` button style should exact like [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) on `Log In` page. Please implment my suggestions 🙏🏼 `[MOBILE VERSION]`

image

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<div class="container mt-4">
+<h3 class="category">CATEGORY</h3>
```
Comment (mukhammadrizooff) [inline app/views/groups/index.html.erb]: - [ ] [REQUIRED] Good job so far! Please give exact design as [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) to your `Home` page's all elements so you should give correct style to `Navbar` becasue there disappear `Log Out` button and elements locate opposite side and created `categories` texts appear big and also compositions are bit wrong placed, lastly give correct design to `Add New Category` button 🙏🏼    `[MOBILE VERSION]`

image

File: app/views/expenses/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<div class="container mb-3">
+  <div class="expense-header">
+     <h1> <%= link_to user_groups_path do %>
+        <i class="fa fa-arrow-circle-left" aria-hidden="true" style="color: #3778c2"> </i>
+        <% end %>
+     </h1>
+  </div>
+  <h3 class="d-flex justify-content-center align-items-center expense">TRANSACTONS</h3>
```
Comment (mukhammadrizooff) [inline app/views/expenses/index.html.erb]: - [ ] [REQUIRED] Good job so far! Please fix `back` button's location and give style to details(`transactions`) part as [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) and also change design to `Add Transaction` button to [correct one](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) 🙏🏼 `[MOBILE VERSION]`

image

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @YoutLone, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- All Tests passed✔️ 
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Htet, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**